modules/criteoBidAdapter.js

```diff
@@ -0,0 +1,279 @@
+import { loadScript } from 'src/adloader';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { EVENTS } from 'src/constants';
+import { parse } from 'src/url';
+import * as utils from 'src/utils';
+
+var events = require('src/events');
+
+const ADAPTER_VERSION = 4;
+const BIDDER_CODE = 'criteo';
+const CDB_ENDPOINT = '//bidder.criteo.com/cdb';
+const INTEGRATION_MODES = {
+  'amp': 1,
+};
+const PROFILE_ID = 207;
+
+// Unminified source code can be found in: https://github.com/Prebid-org/prebid-js-external-js-criteo/blob/master/dist/prod.js
+const PUBLISHER_TAG_URL = '//static.criteo.net/js/ld/publishertag.prebid.js';
+
+/** @type {BidderSpec} */
+export const spec = {
+  code: BIDDER_CODE,
+
+  /**
+   * @param {object} bid
+   * @return {boolean}
+   */
+  isBidRequestValid: bid => (
+    !!(bid && bid.params && (bid.params.zoneId || bid.params.networkId))
+  ),
+
+  /**
+   * @param {BidRequest[]} bidRequests
+   * @param {*} bidderRequest
+   * @return {ServerRequest}
+   */
+  buildRequests: (bidRequests, bidderRequest) => {
+    let url;
+    let data;
+
+    // If publisher tag not already loaded try to get it from fast bid
+    if (!publisherTagAvailable()) {
+      window.Criteo = window.Criteo || {};
+      window.Criteo.usePrebidEvents = false;
+
+      tryGetCriteoFastBid();
+
+      // Reload the PublisherTag after the timeout to ensure FastBid is up-to-date and tracking done properly
+      setTimeout(() => {
+        loadScript(PUBLISHER_TAG_URL);
```

Comment:
We are going to update this function signature to include module/bidder code so we can track who is loading external JS. Will reply shortly with the changes.

Reply:
@ahubertcriteo @Spark-NF 
Please use `loadExternalScript` method instead.
This will enforce a whitelist of which adapters are allowed to load external JS: 
https://github.com/prebid/Prebid.js/blob/master/src/adloader.js#L9